include/Dialect/Poly/IR/PolyOps.td

```diff
@@ -65,13 +65,13 @@ def Poly_GetCoeffOp : Poly_Op<"get_coeff"> {
   let summary = "Gets the coefficient of X^{index} from a polynomial.";
 
   let arguments = (ins
-    Polynomial:$x,
+    PolynomialLike:$x,
     Index:$index
   );
 
   // NOTE: This assumes the underlying type is an integer.
   let results = (outs
-    AnyInteger:$output
+    AnyIntegerLike:$output
```

Comment:
I think I agree with @asraa here, having `poly.get_coeff` over tensors of polynomials might not be needed at the moment but adds a bunch of complexity both for dialects lowering to poly and for dialects lowering from poly.

Reply:
If we did define it over tensors, wouldn't it then also be `ElementwiseMappable`? Assuming that's the semantics we wanted from that! If passing in tensors of polynomials, one might also reasonably replace the index with a shape and simply apply the first k-1 dimensions to the container/tensor and the last one to the polynomial itself....

I'm not suggesting we do that, just highlighting how having tensor inputs here might be a bit confusing as it's not clear what the semantics should be.

Reply:
Happy to take a simple road, we can make it `ElementwiseMappable` in the future.